Add --include-untracked flag to stash show command

Golim opened the pull request:
- **PR Description**

This PR fixes #3931 by adding `--include-untracked` to `git stash show` command. Untracked stashed file diff is now shown in the main view.

![before this PR](https://github.com/user-attachments/assets/81097c71-3764-4467-a960-a87c51921139)
*Before: the untracked stashed file is not shown in lazygit*.

![after this PR](https://github.com/user-attachments/assets/4aafd5f6-5305-4f9b-9068-1570d22e755c)
*After: the untracked stashed file diff is shown in lazygit*.

## One Issue

This PR does not fix the fact that untracked stashed files are not shown in the list of files in the "Stash" panel when entering a stash entry, I could not find an easy way to make them appear there.

* [x] Cheatsheets are up-to-date (run `go generate ./...`)
* [x] Code has been formatted (see [here](https://github.com/jesseduffield/lazygit/blob/master/CONTRIBUTING.md#code-formatting))
* [x] Tests have been added/updated (see [here](https://github.com/jesseduffield/lazygit/blob/master/pkg/integration/README.md) for the integration test guide)
* [x] Text is internationalised (see [here](https://github.com/jesseduffield/lazygit/blob/master/CONTRIBUTING.md#internationalisation))
* [ ] If a new UserConfig entry was added, make sure it can be hot-reloaded (see [here](https://github.com/jesseduffield/lazygit/blob/master/docs/dev/Codebase_Guide.md#using-userconfig))
* [ ] Docs have been updated if necessary
* [x] You've read through your own file changes for silly mistakes etc

jesseduffield reviewed (approved):
LGTM

jesseduffield commented:
Sorry for late review. I don't think this needs a test: it's not adding any conditional logic

jesseduffield commented:
Spoke too soon, turns out updating tests is mandatory because currently they're breaking. Are you up to fix those tests @Golim ?

Golim commented:
`git stash show --include-untracked` fails in older git versions with the following output: `usage: git diff [<options>] [<commit> [<commit>]] [--] [<path>...]`

Using `-u` works correctly also in older git versions, so I replaced `--include-untracked` with `-u` in the `git stash show` command. I left a comment in the code to signal why I did not use the long version of the argument.

I also updated `TestStashStashEntryCmdObj` to expect `-u` in the command arguments

All tests pass on my machine now, also using older git versions
